ui/components/app/signature-request/signature-request.component.js

```diff
@@ -222,20 +224,22 @@ export default class SignatureRequest extends PureComponent {
           >
             {this.context.t('signatureRequestGuidance')}
           </Typography>
-          <div>
-            <Button
-              type="link"
-              onClick={() => this.setState({ showContractDetails: true })}
-              className="signature-request-content__verify-contract-details"
-            >
-              <Typography
-                variant={TYPOGRAPHY.H7}
-                color={COLORS.PRIMARY_DEFAULT}
+          {verifyingContract ? (
+            <div>
+              <Button
+                type="link"
+                onClick={() => this.setState({ showContractDetails: true })}
+                className="signature-request-content__verify-contract-details"
               >
-                {this.context.t('verifyContractDetails')}
-              </Typography>
-            </Button>
-          </div>
+                <Typography
+                  variant={TYPOGRAPHY.H7}
+                  color={COLORS.PRIMARY_DEFAULT}
+                >
+                  {this.context.t('verifyContractDetails')}
+                </Typography>
+              </Button>
+            </div>
+          ) : null}
```

Comment:
Nit: will be nice to have this condition also tested.

Reply:
@jpuri I have added a test for this. cc: @chloeYue